### PR TITLE
Add full RO pool support for SQLite

### PR DESCRIPTION
**Pull request by hifi**

### CODE REVIEW

**Comment by hifi** [comment]
Made this more generic by allowing user configured drivers to be used so our downstream test in Hungryserv can set the busy timeout to zero making unit tests non-blocking without compromising on the test scope.

Downstream PR depends on this to be merged first to make the tests pass.

**Comment by hifi** [comment]
Rejigged based on our conversation.

File: util/dbutil/database.go [hunk 1]
```diff
@@ -186,54 +194,87 @@ type Config struct {
...
+		if len(uriParts) == 2 {
+			var err error
+			qs, err = url.ParseQuery(uriParts[1])
+			if err != nil {
+				return nil, err
+			}
+
+			qs.Del("_txlock")
+		}
+		qs.Set("_query_only", "true")
```
Comment (tulir) [inline util/dbutil/database.go:266]: The uri modification should probably only happen if the read-only pool uri is empty

File: util/dbutil/database.go [hunk 2]
```diff
@@ -186,54 +194,87 @@ type Config struct {
...
-		return nil, err
-	}
-	conn, err := sql.Open(cfg.Type, cfg.URI)
+	wrappedDB, err := NewWithDialect(cfg.URI, cfg.Type)
 	if err != nil {
 		return nil, err
 	}
 	if logger == nil {
 		logger = NoopLogger
 	}
```
Comment (tulir) [inline util/dbutil/database.go:241]: This can be simplified to only change `wrappedDB.Log` when `logger != nil`

File: util/dbutil/database.go [hunk 3]
```diff
@@ -172,10 +174,16 @@ func NewWithDialect(uri, rawDialect string) (*Database, error) {
...
 		return nil, err
 	}
-	return NewWithDB(db, rawDialect)
+
+	wrappedDB, err := NewWithDB(db, rawDialect)
+	if err != nil {
+		return nil, err
+	}
+
+	return wrappedDB, nil
```
Comment (tulir) [inline util/dbutil/database.go:183]: This change is now unnecessary

**Review by tulir — APPROVED** [review]

**Pull request merged**